src/Rings/mpoly-localizations.jl

```diff
@@ -1435,6 +1435,24 @@ function coordinates(a::RingElem, I::MPolyLocalizedIdeal; check::Bool=true)
   end
 end
 
+function coordinates(
+    a::RingElem, I::MPolyLocalizedIdeal{LRT}; check::Bool=true
+  ) where {LRT<:MPolyLocalizedRing{<:Any, <:Any, <:Any, <:Any, <:MPolyPowersOfElement}}
+  L = base_ring(I)
+  parent(a) === L || return coordinates(L(a), I, check=check)
```

Comment:
In sage/python silent conversions are the way to go. But in Oscar we want explicit conversions only.
Silent conversions can be convenient in interactive coding but when doing anything serious it is much better to 
get an error than a bug that is covered up because of some conversion (and then the coverup often (silently!) produces wrong results).
```suggestion
  parent(a) === L || error("element and ideal are not defined over the same ring")```

Reply:
I see what you mean. But other people have different opinions on this and I am a bit tired to go back and forth with this. For instance, Wolfram was once wondering why `1 in I` was not working. And it wasn't because the implicit cast had not been there, so I added it. I would be happy if this was once clarified and written down and then I can stick to it. But until then, I see no point in pursuing such discussions here.

Reply:
Whether it is "dangerous" depends on what `I` is. If `I` is an ideal over $\mathbb{F_2}[x,y]$. Then $1\ in I$ is even mathematically wrong.  (I could live with silent convertions of `1` to rings of characteristic 0).

But as far as I can tell `1 in I` ist still not accepted anywhere
```
julia> k = GF(2)
Galois field with characteristic 2

julia> P,(x,y) = PolynomialRing(k,[:x,:y])
(Multivariate Polynomial Ring in x, y over Galois field with characteristic 2, gfp_mpoly[x, y])

julia> I = ideal(P,[1])
ideal(1)

julia> 1 in I
ERROR: MethodError: no method matching iterate(::MPolyIdeal{gfp_mpoly})
......

julia> P,(x,y) = PolynomialRing(QQ,[:x,:y])
(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])

julia> I = ideal(P,[1])
ideal(1)

julia> 1 in I
ERROR: MethodError: no method matching iterate(::MPolyIdeal{fmpq_mpoly})
```
But you are right that this is not the right place for this discussion